src/index.js

```diff
@@ -368,10 +370,14 @@ UseWalletProvider.defaultProps = {
   pollBlockNumberInterval: 5000,
 }
 
-function UseWalletProviderWrapper(props) {
+function UseWalletProviderWrapper({ ethereum: providedEthereum, ...props }) {
+  const getLibrary = useCallback(ethereum => providedEthereum || ethereum, [
+    providedEthereum,
+  ])
+
   return (
-    <Web3ReactProvider getLibrary={ethereum => ethereum}>
-      <UseWalletProvider {...props} />
+    <Web3ReactProvider getLibrary={getLibrary}>
+      <UseWalletProvider providedEthereum={providedEthereum} {...props} />
```

Comment:
We now need to drill this prop into the `UseWalletProvider` component above so we can properly instantiate the `InjectedConnector`, as we can't use the `library` prop from `Web3ReactProvider` as its `undefined` until the user activates the account.

Reply:
Should we just name this `ethereum`? I think passing one already implies that it is provided by the user.

Reply:
Ah forget this, I thought we were on the external API (`UseWalletProviderWrapper`).